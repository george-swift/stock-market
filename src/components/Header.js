import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  FaHome, FaChartLine, FaInfoCircle, FaTrademark,
} from 'react-icons/fa';

const Header = ({ reset }) => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
    <div className="container-fluid">
      <Link to="/" className="navbar-brand" onClick={() => reset()}>
        <span className="me-2">Stock Market</span>
        <FaTrademark />
      </Link>
      <button
        type="button"
        className="navbar-toggler"
        data-bs-toggle="collapse"
        data-bs-target="#fmNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navbar"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="fmNav">
        <ul className="navbar-nav">
          <li className="nav-item">
            <Link to="/" className="nav-link" onClick={() => reset()}>
              <FaHome />
              <span className="ms-2">Home</span>
            </Link>
          </li>
          <li className="nav-item">
            <Link to="/listings" className="nav-link" onClick={() => reset()}>
              <FaChartLine />
              <span className="ms-2">Stock List</span>
            </Link>
          </li>
          <li className="nav-item">
            <Link to="/about" className="nav-link" onClick={() => reset()}>
              <FaInfoCircle />
              <span className="ms-2">About</span>
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

Header.propTypes = { reset: PropTypes.func.isRequired };

export default Header;
