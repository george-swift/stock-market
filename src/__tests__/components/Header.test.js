import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, within, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../../components/Header';

const reset = jest.fn(() => {});

describe('Snapshot of the Header component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header reset={reset} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Querying the Header component', () => {
  beforeEach(() => render(
    <Router>
      <Header reset={reset} />
    </Router>,
  ));

  it('should have a link to the home page in brand name', () => {
    const brandName = document.querySelector('.navbar-brand');
    expect(brandName).toBeInTheDocument();
    fireEvent.click(brandName);
    expect(reset).toHaveBeenCalledTimes(1);
    const app = within(brandName).getByText('Stock Market');
    expect(app).toBeInTheDocument();
    expect(brandName).toHaveAttribute('href', '/');
  });

  it('should have hamburger for mobile devices', () => {
    const navBar = document.querySelector('nav');
    const hamburger = within(navBar).getByRole('button');
    expect(hamburger).toHaveClass('navbar-toggler');
  });

  it('should have links to the three main SPA pages', () => {
    const navBar = document.querySelector('.navbar-nav');

    const routes = navBar.querySelectorAll('.nav-link');
    expect(routes).toHaveLength(3);

    routes.forEach((link) => fireEvent.click(link));
    expect(reset).toHaveBeenCalledTimes(3);

    const homePage = within(navBar).getByText('Home');
    expect(homePage).toBeInTheDocument();

    const stockListPage = within(navBar).getByText('Stock List');
    expect(stockListPage).toBeInTheDocument();

    const aboutPage = within(navBar).getByText('About');
    expect(aboutPage).toBeInTheDocument();
  });
});
