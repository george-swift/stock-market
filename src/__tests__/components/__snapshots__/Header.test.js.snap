// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Header component should render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <span
        className="me-2"
      >
        Stock Market
      </span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M260.6 96H12c-6.6 0-12 5.4-12 12v43.1c0 6.6 5.4 12 12 12h85.1V404c0 6.6 5.4 12 12 12h54.3c6.6 0 12-5.4 12-12V163.1h85.1c6.6 0 12-5.4 12-12V108c.1-6.6-5.3-12-11.9-12zM640 403l-24-296c-.5-6.2-5.7-11-12-11h-65.4c-5.1 0-9.7 3.3-11.3 8.1l-43.8 127.1c-7.2 20.6-16.1 52.8-16.1 52.8h-.9s-8.9-32.2-16.1-52.8l-43.8-127.1c-1.7-4.8-6.2-8.1-11.3-8.1h-65.4c-6.2 0-11.4 4.8-12 11l-24.4 296c-.6 7 4.9 13 12 13H360c6.3 0 11.5-4.9 12-11.2l9.1-132.9c1.8-24.2 0-53.7 0-53.7h.9s10.7 33.6 17.9 53.7l30.7 84.7c1.7 4.7 6.2 7.9 11.3 7.9h50.3c5.1 0 9.6-3.2 11.3-7.9l30.7-84.7c7.2-20.1 17.9-53.7 17.9-53.7h.9s-1.8 29.5 0 53.7l9.1 132.9c.4 6.3 5.7 11.2 12 11.2H628c7 0 12.5-6 12-13z"
        />
      </svg>
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navbar"
      className="navbar-toggler"
      data-bs-target="#fmNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="fmNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              />
            </svg>
            <span
              className="ms-2"
            >
              Home
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/listings"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
              />
            </svg>
            <span
              className="ms-2"
            >
              Stock List
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              />
            </svg>
            <span
              className="ms-2"
            >
              About
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
